.github/workflows/C_linter.yaml

```diff
@@ -0,0 +1,24 @@
+on:
```

Comment:
how about we run on all pushes and pull requests. why not?

Reply:
Re: push pull requests - you mean regardless of the branch? We can do that, as it'll be useful if anyone wants to develop on another branch and see the output of the workflows.